G1ANT.Addon.Selenium/Api/SeleniumWrapper.cs

```diff
@@ -393,22 +386,50 @@ public string GetOuterHtml(SeleniumCommandArguments search)
 
         public string GetTextValue(SeleniumCommandArguments search, TimeSpan timeout)
         {
-            PreCheckCurrentWindowHandle();
-            if (!string.IsNullOrEmpty(search.IFrameSearch?.Value))
-                webDriver.SwitchTo().Frame(FindElement(search.IFrameSearch.Value, search.IFrameBy.Value, timeout));
-            var element = FindElement(search.Search.Value, search.By.Value, timeout);
+            var element = GetElementInFrame(search, timeout);
+
             var res = element?.Text ?? string.Empty;
             if (!string.IsNullOrEmpty(search.IFrameSearch?.Value))
                 webDriver.SwitchTo().DefaultContent();
             return res;
         }
 
-        public void SetAttributeValue(string attributeName, string attributeValue, SeleniumCommandArguments search, TimeSpan timeout)
+        public DataTable GetTableElement(SeleniumCommandArguments search, TimeSpan timeout)
+        {
+            var element = GetElementInFrame(search, timeout);
+
+            if (element == null || element.TagName != "table")
```

Comment:
Split this expression into two separate and throw different exception if tagname is not "table". It will give better error message to the RPA developer